### ISSUE TITLE
Consensus-Full-Nodes - resources usage issue

### ISSUE THREAD

**Issue by tty47** [celestiaorg/celestia-app#2935]
### Summary of Bug

Hello team! 👋 

I want to report to you an issue that we are facing with the consensus-full-nodes, we have been facing issues when trying to run the nodes with less than 20GB of RAM, by the time the nodes have to sync the chain (for example in mocha) the nodes cannot sync with less than this amount of resources. Even if we set the maximum to 20GB they are trying to get all the resources of the server until they get OOM and crash.
This happens if the nodes have to sync either from scratch or since some days ago.

I consider, that they should work with the resources that they have, even if that means that it's going to take longer to sync, but crashing because of that, might be an issue.

![resources-usage](https://github.com/celestiaorg/celestia-app/assets/32740567/76001e24-9b52-4af8-a6cb-25de363e7463)

I assume that the nodes should work even with 8GB, as we have [here](https://docs.celestia.org/nodes/overview#overview-to-running-nodes-on-celestia)

cc: @celestiaorg/devops 

### Version

`v1.3.0`

### Steps to Reproduce

Start a consensus-full-node and connect it to an existing chain (mocha for example), then, it will have to sync to catch up.

### For Admin Use

- [ ] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate contributors tagged
- [ ] Contributor assigned/self-assigned

**Comment by rootulp** [comment]
Thanks for the issue! A few questions:

1. Based on the screenshot: do you have any idea on why this is happening for consensus-full-snapshot-0 and not for consensus-full-2 or consensus-full-3? How does conensus-full-snapshot-0 differ from the other consensus nodes? Perhaps a config change?
2. Does this repro on other chains besides mocha-4?

**Comment by evan-forbes** [comment]
I think we ran into this before, and becuase of all of the ibc memo spam on mocha, the tx-index will eat gobs of memory. Was the tx-index configured to be on @jrmanes ?

**Comment by rootulp** [comment]
Based on the lines you linked, it looks like the tx indexer is enabled and set to `kv` which uses goleveldb. If the consensus nodes don't need to index transactions, we can remove those lines because the default is `null` which disables tx indexing. If we do need tx indexing, we can explore alternative db_backends and/or the psql option.

**Comment by evan-forbes** [comment]
if this is purely related to the kv indexer, can we perhaps close this issue and open a new one to improve the KV?

**Comment by evan-forbes** [comment]
we should be able to close this issue once we are able to run v2 in production, as that includes a new version of the kv that should remedy the massive amounts of memory used by the existing kv store

**Comment by rootulp** [comment]
Nina backported https://github.com/celestiaorg/celestia-core/pull/1405 to celestia-core [v1.38.0-tm-v0.34.29](https://github.com/celestiaorg/celestia-core/releases/tag/v1.38.0-tm-v0.34.29) which was released in celestia-app [v1.13.0](https://github.com/celestiaorg/celestia-app/releases/tag/v1.13.0). Rachid bumped celestia-node to that release in [this PR](https://github.com/celestiaorg/celestia-node/pull/3470).

TLDR: we don't need to wait until celestia-app v2 is running in production. As soon as celestia-node cuts a release from main (likely v0.15.0), we can use the lightweight tx status work.
